AppCenter/AppCenter/Internals/Channel/MSChannelGroupDefault.m

```diff
@@ -54,6 +55,15 @@ - (instancetype)initWithIngestion:(nullable MSAppCenterIngestion *)ingestion {
   return channel;
 }
 
+- (id<MSChannelUnitProtocol>)channelUnitForGroupId:(NSString *)groupId {
+  for (MSChannelUnitDefault *channel in self.channels) {
```

Comment:
nit: any reason its not a dictionary and just look up by key?

Reply:
Low cost since most of cases are iterating all channels but definitely we can improve it.
For now, we have only one case to get channelUnit, called only once, and it is being kept in the memory so I'll leave it.